ui-lib/src/components/Layout/SideBar/SideBar.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useState, createContext } from "react";
+import styled from "styled-components";
+import Flex from "@ui/components/Layout/Flex";
+import SideBarItem from "./components/SideBarItem/SideBarItem";
+import SideBarIcon from "./components/SideBarIcon/SideBarIcon";
+import SideBarToggle from "./components/SideBarToggle/SideBarToggle";
+import { CSSTransition } from "react-transition-group";
+
+const Nav = styled(Flex)`
+  position: relative;
+  padding: 68px 12px 0;
+  row-gap: 1.5rem;
+  height: 100vh;
+  max-width: 14.875rem;
+  color: ${props => props.theme.colors.palette.v2.text.default};
+  border-right: 1px solid ${props => props.theme.colors.palette.v2.grey.border};
+  background-color: ${props => props.theme.colors.palette.v2.background.paper};
+  transition: max-width 300ms;
+  will-change: max-width;
+
+  &.nav-enter {
+    max-width: 75px;
+  }
+  &.nav-enter-done {
+    max-width: 14.875rem;
+  }
+  &.nav-exit {
+    max-width: 14.875rem;
+  }
+  &.nav-exit-done {
+    max-width: 75px;
+  }
+`;
+
+export type SideBarProps = {
+  children: Array<JSX.Element>;
+  onToggle: () => void;
+  isExpanded?: boolean;
+};
+
+type SideBarContextType = { isExpanded: boolean; onToggle: () => void };
+export const SideBarContext = createContext<Partial<SideBarContextType>>({});
+
+const SideBar = ({ children, onToggle, isExpanded = true }: SideBarProps): JSX.Element => {
+  const [isToggleDisplayed, setToggleDisplayed] = useState(false);
+
+  return (
+    <SideBarContext.Provider value={{ isExpanded, onToggle }}>
```

Comment:
Not a big deal, but it might be good for performance reasons to memoize the provider value. Otherwise it will change the reference of the object every time the component gets re-rendered and force the rendering of every child consumer.

Could be something like:

```ts
const providerValue = useMemo(() => ({ isExpanded, onToggle }), [isExpanded, onToggle]);
```

I made some captures to show the difference on mouse enter / leave

https://user-images.githubusercontent.com/86958797/129918307-6a51f75c-957e-4792-b174-35448259b4cd.mp4

https://user-images.githubusercontent.com/86958797/129918299-1fe00edd-3bc8-49c8-92ee-5ec6302b0af6.mp4

Reply:
Thanks for your suggestion, I used it 👍